output/mistral_fluentbit/mistral_fluentbit.c

```diff
@@ -0,0 +1,553 @@
+ * the common characters that need special treatment as they are encountered.
+ * Once the copy is complete the memory is reallocated to reduce wastage.
+ *
+ * Parameters:
+ *   string - The string whose content needs to be escaped
+ *
+ * Returns:
+ *   A pointer to newly allocated memory containing the escaped string or
+ *   NULL on error
+ */
+static char *fluentbit_escape(const char *string)
+{
+    if (!string) {
+        return NULL;
+    }
+    size_t len = strlen(string);
+
+    char *escaped = calloc(1, (2 * len + 1) * sizeof(char));
+    char *p, *q;
+    if (escaped) {
+        for (p = (char *)string, q = escaped; *p; p++, q++) {
+            switch (*p) {
+            case '"':
+                *q++ = '\\';
+                *q = '"';
+                break;
+            case '\\':
+                *q++ = '\\';
+                *q = '\\';
+                break;
+            case '\b':
+                *q++ = '\\';
+                *q = 'b';
+                break;
+            case '\f':
+                *q++ = '\\';
+                *q = 'f';
+                break;
+            case '\n':
+                *q++ = '\\';
+                *q = 'n';
+                break;
+            case '\r':
+                *q++ = '\\';
+                *q = 'r';
+                break;
+            case '\t':
+                *q++ = '\\';
+                *q = 't';
+                break;
+            default:
+                *q = *p;
+                break;
+            }
+        }
+        /* Memory was allocated with calloc so string is already null terminated */
+        char *small_escaped = realloc(escaped, q - escaped + 1);
+        if (small_escaped) {
+            return small_escaped;
+        } else {
+            return escaped;
+        }
+    } else {
+        return NULL;
+    }
+}
+
+/*
+ * mistral_startup
+ *
+ * Required function that initialises the type of plug-in we are running. This
+ * function is called immediately on plug-in start-up.
+ *
+ * In addition this plug-in needs to initialise the Fluent Bit connection
+ * parameters. The stream used for error messages defaults to stderr but can
+ * be overridden here by setting plugin->error_log.
+ *
+ * Parameters:
+ *   plugin - A pointer to the plug-in information structure. This function
+ *            must set plugin->type before returning, if it doesn't the plug-in
+ *            will immediately shut down.
+ *   argc   - The number of entries in the argv array
+ *   argv   - A pointer to the argument array passed to main.
+ *
+ * Returns:
+ *   void - but see note about setting plugin->type above.
+ */
+void mistral_startup(mistral_plugin *plugin, int argc, char *argv[])
+{
+    /* Returning without setting plug-in type will cause a clean exit */
+
+    static const struct option options[] = {
+        {"error", required_argument, NULL, 'e'},
+        {"host", required_argument, NULL, 'h'},
+        {"port", required_argument, NULL, 'p'},
```

Comment:
Does this mean that var/v is not yet supported?  This is something that we recommend people use - to get environment variables - such as $USER and any project/group identifiers that may be set.

Reply:
I have added and tested var support.